web/packages/design/src/utils/useResizeObserver.ts

```diff
@@ -16,46 +16,70 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import { RefObject, useCallback, useLayoutEffect } from 'react';
+import {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  type RefCallback,
+} from 'react';
 
 /**
- * useResizeObserver sets up a ResizeObserver for ref and calls callback on each resize.
+ * useResizeObserver creates a ResizeObserver which fires a callback
+ * when the provided element is resized. The callback is called with the
+ * ResizeObserverEntry. The observer is disconnected when the element is
+ * unmounted.
  *
- * It does not fire if ref.current.contentRect.height is zero, to account for a special case in
- * Connect where tabs are hidden using `display: none;`.
+ * Returns a ref callback to attach to the element to be observed – the element
+ * must be a HTMLElement but may be conditionally null.
```

Comment:
> but may be conditionally null

Please explain in comment in what case it could be null?

Reply:
I believe it's meant as an info to the caller that the ref supports conditional rendering, as in you're free to write code like this:

https://github.com/gravitational/teleport/blob/c684051efadf3470e8a1e3bc8393aed4c70012e2/web/packages/design/src/utils/useResizeObserver.test.tsx#L79

The previous version of the hook didn't quite support that.